apkid/rules/dex/obfuscators.yara

```diff
@@ -241,3 +242,24 @@ rule aamo_str_enc : obfuscator
   condition:
     1 of ($opcodes*) and all of ($a, $b)
 }
+
+rule appsuit : obfuscator
+{
+    meta:
+        description = "AppSuit"
+        url         = "http://www.stealien.com/appsuit.html"
+        sample      = "b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328"
+        author      = "Eduardo Novella"
+
+    strings:
+        $class1      = "Lcom/stealien/const"
+        $class2      = "Lcom/stealien/appsuit"
+        $class3      = "La_lock"
+        $native_lib  = "AppSuit"
+        $package     = "stealien"
+        $packer      = "AppSuitDexLoader"
+
+    condition:
+        2 of them and not is_apk
```

Comment:
Rather than `not is_apk`, why not `is_dex` and remove the include line for `apk/common`?

Reply:
`not is_apk` can be either dex or elf. The issue here is that the `/` in the pkg name wont match in ELF files.

Reply:
Sounds like you could explicitly bunch the rules then? E.g.;
```
conditions:
  (is_dex and $dex_string) or (is_elf and $elf_string)
```

Maye this seems redundant, however it would show you what is hitting why a bit easier and never fp?

Reply:
Done!

Reply:
Ops, I committed before reading your suggestion. It seems OK now.


```sh
[20:43 edu@iloveyou appsuit] >  for p in `ls`; do echo -ne "\n\n$p\n"; apkid $p;done


11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] 11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, network operator name check
 |-> compiler : dexlib 2.x
[*] 11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0


147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, possible Build.SERIAL check
 |-> compiler : dexlib 2.x
[*] 147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7


2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024!classes.dex
 |-> compiler : dexlib 2.x
[*] 2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024


3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] 3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, network operator name check, possible vm check
 |-> compiler : dexlib 2.x
[*] 3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289


6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea!classes2.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, subscriber ID check
 |-> compiler : dx
[*] 6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea!classes.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, network operator name check, possible vm check
 |-> compiler : dx
[*] 6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea


73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.FINGERPRINT check, Build.MANUFACTURER check, Build.TAGS check, SIM operator check, network operator name check, possible Build.SERIAL check, ro.product.device check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] 73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2!classes.dex
 |-> anti_vm : Build.BRAND check, Build.DEVICE check, Build.FINGERPRINT check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, network operator name check, possible Build.SERIAL check
 |-> compiler : dexlib 2.x
[*] 73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2


73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, device ID check, emulator file check, network operator name check, possible Build.SERIAL check, possible vm check, ro.kernel.qemu check, ro.product.device check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] 73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] 73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84


824054dccad61952d48752c4176640e8589bc18617d1d5442307444367f707b3
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 824054dccad61952d48752c4176640e8589bc18617d1d5442307444367f707b3!classes.dex
 |-> compiler : dexlib 2.x
[*] 824054dccad61952d48752c4176640e8589bc18617d1d5442307444367f707b3


b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a!classes2.dex
 |-> compiler : dx
[*] b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a!classes.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, network operator name check, possible vm check
 |-> compiler : dx
[*] b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a


c0812c74fac6ebda16fc06686dfc4a921f4045d37eae6a9a098849fd7315c17b
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] c0812c74fac6ebda16fc06686dfc4a921f4045d37eae6a9a098849fd7315c17b!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, network operator name check, possible Build.SERIAL check
 |-> compiler : dexlib 2.x
[*] c0812c74fac6ebda16fc06686dfc4a921f4045d37eae6a9a098849fd7315c17b


e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d!classes.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, network operator name check, possible Build.SERIAL check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d


e96895d7cd7e5d69ccd5d0690df6eae94d3f62a6410e12ca8dc654664e2155d5
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] e96895d7cd7e5d69ccd5d0690df6eae94d3f62a6410e12ca8dc654664e2155d5!classes.dex
 |-> anti_vm : possible vm check
 |-> compiler : dexlib 2.x
[*] e96895d7cd7e5d69ccd5d0690df6eae94d3f62a6410e12ca8dc654664e2155d5


ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, network operator name check
 |-> compiler : dexlib 2.x
[*] ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a
[20:44 edu@snps appsuit] >  for p in `ls`; do echo -ne "\n\n$p\n"; apkid $p;done


11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, network operator name check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 11b630811606216d89e2d9ef4de4a6f4ae053724a964ab0b18251efc5236ccf0


147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, possible Build.SERIAL check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7
 |-> packer : AppSuit


2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024!classes.dex
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024
 |-> packer : AppSuit


3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, network operator name check, possible vm check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 3b2db8c9a582a06c509bc471406ed641567dcbc7d2511bd9c1a460339f541289


6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea!classes2.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, subscriber ID check
 |-> compiler : dx
 |-> obfuscator : AppSuit
[*] 6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea!classes.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, network operator name check, possible vm check
 |-> compiler : dx
[*] 6055deceb83233cceefc89b2bce4e978fd417820c5f534b0df66415122f394ea


73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.FINGERPRINT check, Build.MANUFACTURER check, Build.TAGS check, SIM operator check, network operator name check, possible Build.SERIAL check, ro.product.device check, subscriber ID check
 |-> compiler : dexlib 2.x
[*] 73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2!classes.dex
 |-> anti_vm : Build.BRAND check, Build.DEVICE check, Build.FINGERPRINT check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, network operator name check, possible Build.SERIAL check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 73adddc407297c7545a8d9cd29462267f15806453e6112ff6558b9678b2f8aa2
 |-> packer : AppSuit


73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, device ID check, emulator file check, network operator name check, possible Build.SERIAL check, possible vm check, ro.kernel.qemu check, ro.product.device check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 73dddba334fe1d19504a160c3ed87e47d796867df2c0766b3cdf4ef6a4239f84


824054dccad61952d48752c4176640e8589bc18617d1d5442307444367f707b3
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] 824054dccad61952d48752c4176640e8589bc18617d1d5442307444367f707b3!classes.dex
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] 824054dccad61952d48752c4176640e8589bc18617d1d5442307444367f707b3
 |-> packer : AppSuit


b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a!classes2.dex
 |-> compiler : dx
[*] b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a!classes.dex
 |-> anti_debug : Debug.isDebuggerConnected() check
 |-> anti_vm : Build.FINGERPRINT check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, network operator name check, possible vm check
 |-> compiler : dx
 |-> obfuscator : AppSuit
[*] b8184414f136de13f08da578c3de49aa536c02d20eaa8ca467413729d12f352a


c0812c74fac6ebda16fc06686dfc4a921f4045d37eae6a9a098849fd7315c17b
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] c0812c74fac6ebda16fc06686dfc4a921f4045d37eae6a9a098849fd7315c17b!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, network operator name check, possible Build.SERIAL check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] c0812c74fac6ebda16fc06686dfc4a921f4045d37eae6a9a098849fd7315c17b
 |-> packer : AppSuit


e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d!classes.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, network operator name check, possible Build.SERIAL check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d


e96895d7cd7e5d69ccd5d0690df6eae94d3f62a6410e12ca8dc654664e2155d5
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] e96895d7cd7e5d69ccd5d0690df6eae94d3f62a6410e12ca8dc654664e2155d5!classes.dex
 |-> anti_vm : possible vm check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] e96895d7cd7e5d69ccd5d0690df6eae94d3f62a6410e12ca8dc654664e2155d5
 |-> packer : AppSuit


ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a
[+] APKiD 1.2.1 :: from RedNaga :: rednaga.io
[*] ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a!classes2.dex
 |-> anti_vm : Build.BOARD check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, SIM operator check, device ID check, network operator name check, possible Build.SERIAL check, possible vm check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a!classes.dex
 |-> anti_vm : Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, network operator name check
 |-> compiler : dexlib 2.x
 |-> obfuscator : AppSuit
[*] ec1fa5081267291d75449059d83836dcec3bcdf593d2fe2c3635a897a42ff64a
```

Reply:
But the order of printing results is odd IMHO @CalebFenton  @strazzere

Reply:
What is the odd bit or the anticipated outcome?

Currently it unzips, scans assets, alerts on those assets if a rule hits, then after scanning everything scans the container and alerts on that combined asset.

It seems to make sense to me?

E.g. - this allows it to find multiple things inside a container, then tell you the container itself was protected with blah.

Reply:
Got used to the say before how the apk was packed instead of saying at the end. Nevermind.